### PR TITLE
add ROI pooling layer to the CUDA backend

### PR DESCRIPTION
**Pull request by YashasSamaga**
### This pullrequest changes
- adds ROI pooling layer to the CUDA backend

<cut/>

```
force_builders=Custom
buildworker:Custom=linux-4
docker_image:Custom=ubuntu-cuda:18.04
```

### CODE REVIEW

File: modules/dnn/src/cuda4dnn/primitives/roi_pooling.hpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+#include "../kernels/roi_pooling.hpp"
+
+#include <utility>
+
+namespace cv { namespace dnn { namespace cuda4dnn {
+
+    template <class T>
+    class ROIPoolingOp final : public CUDABackendNode {
+    public:
```
Comment (alalek) [inline modules/dnn/src/cuda4dnn/primitives/roi_pooling.hpp]: Consider avoiding indentation in "whole" file namespaces.

File: modules/dnn/src/cuda4dnn/primitives/roi_pooling.hpp [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        using wrapper_type = GetCUDABackendWrapperType<T>;
+
+        ROIPoolingOp(csl::Stream stream_, float spatial_scale)
```
Comment (alalek) [inline modules/dnn/src/cuda4dnn/primitives/roi_pooling.hpp]: Does `std::move()` used properly below?

Comment (YashasSamaga) [reply modules/dnn/src/cuda4dnn/primitives/roi_pooling.hpp]: Yes, I think. The streams are sharable RAII objects. The `stream_` argument is not a reference. Hence, a copy is created at the site where the constructor is invoked. This copy is moved into `stream` which is a member.

File: modules/dnn/src/cuda4dnn/primitives/roi_pooling.hpp (hunk 1 above)
Comment (YashasSamaga) [reply modules/dnn/src/cuda4dnn/primitives/roi_pooling.hpp]: Should I push a commit now?

**Pull request merged**